### ISSUE TITLE
Feature request: Download by part

### ISSUE THREAD

**Issue by p2s** [Franiac/TwitchLeecher#335]
Hi ! 
Thanks a lot for your great work ! 

I use it a lot while travelling to download league of legend games, but it can last 4-6 hours.
So i usually make part of each video so if a download failed i don't have to wait hours again to download the entire game, I just have to redownload a part of it. 
Another good point is that I can start to watch quickly as soon as the first part is downloaded, while the other part are downloaded in background. And last usage is that i can delete files as soon as i watch them.

But actually it's a bit annoying to do it manually because i need to watch when does the game start, set is as start time, set an end time and remember it. Then doing the second part beginning at the end of first one and so on...

So i imagined two easy way to do it, either by setting length of part wanted : 

![twitchleecher_example_slipt_download](https://user-images.githubusercontent.com/453590/57052753-b22e0780-6ccc-11e9-92de-b29851d15bd8.png)

Or setting the number of parts : 

![twitchleecher_example_slipt_download_nb_part](https://user-images.githubusercontent.com/453590/57052752-b1957100-6ccc-11e9-8819-67244bace07d.png)

And when you click to download it create 4 independents download task in chronological order.

I'm web software developer and i may have some time next weeks, if you seems this feature seems great i will try to do a pull request ;) 

Thanks a lot and do not hesite if you have question :)

**Comment by Franiac** [comment]
It's kinda sad that I have absolutely no time for the new downloader because that would have pause/reconnect capabilities and you would not have to download everything again if it fails.

The split-thingy is a cool idea, but if one part in the middle fails you still need to re-download it manually with the current downloader. The new downloader should also provide a simple "try again" button.

**Comment by AntonAlekseevAA** [comment]
Hello! Sorry for stupid question. Is it Right that this issue is blocked by  https://github.com/Franiac/TwitchLeecher/issues/46 (it is mentioned in  @Franiac  last message), or can be done separately?

Also, if your have some time, can your provide some more info please? Main question - what should be the result? When i confirm download splited by N parts, then in downloads queue should appear N tasks? Then it's runs synchronously (as now)?
In summary. As a result,  need to implement calculation start and end time of each part, create new download request (as if manually added from current UI by user action), end push it to download queue?

Sorry for my english.

**Comment by Codereaders** [comment]
"Pause / Resume" is quite equal to "download by part" from a technical point of view. As both target the download process/ management I wouldn't wonder about #335 is being blocked by #46. But it's only an educated guess, so please don't quote me on that.

**Comment by p2s** [comment]
I almost finish the development of this feature, I will do a pull request tomorrow or friday ;) 
Then we will see if the pull request is accepted until the #46 is implemented :)

**Comment by Franiac** [comment]
#803
